### ISSUE TITLE
Expander leads to infinitive loop in XAML Designer when `RowSpan` property is used

### ISSUE THREAD

**Issue by DrCQ** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3438]
### Bug explanation

I have some `Expander` controls in the main `Grid`. Each `Expander` contains other `Grid`'s with rows and columns. Columns are sharing the size between each other.
```
<Grid.ColumnDefinitions>
    <ColumnDefinition Width="Auto" SharedSizeGroup="A1"/>
    <ColumnDefinition Width="Auto" SharedSizeGroup="A2"/>
    <ColumnDefinition Width="Auto" SharedSizeGroup="A3"/>
    <ColumnDefinition/>
</Grid.ColumnDefinitions>
```
In one of these `Grid` controls, I have an `Image` using the `RowSpan` and `ColumnSpan` property.
```
<Image Grid.Row="0" Grid.RowSpan="6" Grid.Column="0" Grid.ColumnSpan="2" 
       Source="{Binding IconSource, Mode=OneWay}" Width="160" Height="120" HorizontalAlignment="Left"/>
```
This XAML statement leads to an infinity loop in the XAML Designer. The VS is busy, and the Designer shows the content of all columns jumping from the left to the right.
After some tests, I guess the problem is the `RowSpan` property.

Closing VS and reopening the compiled solution seems temporarily to solve the problem, but any attempt to edit the XAML cause the problem again.

### Version

4.9.0

**Comment by Keboo** [comment]
Hi @DrCQ are you able to provide more details or a sample repo? I suspect the issue is with the expander trying to know the size of its container when you have Auto set? But in the code above it is only showing the GridColumns. It would be helpful to see a repo that reproduces the issue.

**Comment by DrCQ** [comment]
Hi @Keboo, 
sorry for the late response, but I was 9 weeks away. Please check [ClearTypeTest](https://github.com/DrCQ/ClearTypeTest).
The clarity problem seems to be solved with version 5.0.1-ci661

![Zoom-100](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/assets/49019093/38a8a07a-1808-4728-b9ed-9a79b9d4d528)
The screenshot shows my `StatusBar `control embedded in Outlined card, plain control and Elevated card.

Also, the 300% zoom looks very similar for all versions of the `StatusBar`.

![Zoom-300](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/assets/49019093/8467f7f4-244d-47f0-ba05-9653869b5a59)

**Comment by DrCQ** [comment]
Sorry, I mixed up two issues - expander loop and the card clarity. The card issue is solved with 5.0.1. 
Concerning the expander, I don't have any examples anymore. I removed the expander from my application due to this behavior.
